polkadot/node/subsystem-bench/src/subsystem-bench.rs

```diff
@@ -222,23 +242,30 @@ impl BenchCli {
 		let mut state = TestState::new(&test_config);
 		let (mut env, _protocol_config) = prepare_test(test_config, &mut state);
 
-		match self.objective {
-			TestObjective::DataAvailabilityRead(_options) => {
-				env.runtime()
-					.block_on(availability::benchmark_availability_read(&mut env, state));
-			},
-			TestObjective::DataAvailabilityWrite => {
-				env.runtime()
-					.block_on(availability::benchmark_availability_write(&mut env, state));
-			},
-			TestObjective::TestSequence(_options) => {},
-		}
+		let usage = match self.objective {
+			TestObjective::DataAvailabilityRead(_options) =>
+				env.runtime().block_on(availability::benchmark_availability_read(
+					"benchmark_availability_read",
+					&mut env,
+					state,
+				)),
+			TestObjective::DataAvailabilityWrite =>
+				env.runtime().block_on(availability::benchmark_availability_write(
+					"benchmark_availability_write",
+					&mut env,
+					state,
+				)),
+			TestObjective::TestSequence(_options) => todo!(),
+		};
 
 		if let Some(agent_running) = agent_running {
 			let agent_ready = agent_running.stop()?;
 			agent_ready.shutdown();
 		}
 
+		let output = if self.ci { serde_yaml::to_string(&vec![usage])? } else { usage.to_string() };
```

Comment:
Nit: This logic is duplicated you can move it in a small funciton.

Reply:
You're right. But we're going to remove the last branch and keep only run with test files. So this duplication is going to be removed as well.